packages/contracts/src/dollar/libraries/LibCurveDollarIncentive.sol

```diff
@@ -159,7 +159,7 @@ library LibCurveDollarIncentive {
         bytes16 res = _one.sub(curPrice.fromUInt()).mul(amount.fromUInt());
         // returns (1- TWAP_Price) * amount.
         return res.div(_one).toUInt();
-        LibTWAPOracle.update();
+        //LibTWAPOracle.update();
```

Comment:
This seems suspicious that its commented out. Should we move this before the return statement?

Reply:
that would get us, bad coverage for some reason, best approach is to refactor the whole function, would you think that's fit for another github issue + including new tests file, specifically for this one

Reply:
it makes sense to me since it's another scope, but let me know! @pavlovcik

Reply:
am +1 in bypass **coveralls.io** , since this whole snippet needs testing

Reply:
Anyway this line is removed as a part of https://github.com/ubiquity/ubiquity-dollar/pull/701

Reply:
let's go!!! @pavlovcik

Reply:
> Anyway this line is removed as a part of https://github.com/ubiquity/ubiquity-dollar/pull/701

You sure? I don't see this file in the changed files list. Besides that doesn't answer the question about if we should retain it before the return statement. I think updating the TWAP seems important because it's the oracle the dictates our entire Dollar's monetary policy.

Reply:
> 

https://github.com/ubiquity/ubiquity-dollar/pull/704#discussion_r1241029672 here

Reply:
this lines mutates the state and it's called many times in the code, refactoring it's suitable here, but not the scope of the PR